src/main/java/org/folio/service/rollover/RolloverValidationService.java

```diff
@@ -56,21 +55,11 @@ private String buildValidateUniquenessQuery(String tenantId) {
       getFullTableName(tenantId, LEDGER_FISCAL_YEAR_ROLLOVER_TABLE));
   }
 
-  private Future<Boolean> isRolloverExists(String query, Tuple parameters, DBClient client) {
+  private Future<Boolean> isRolloverExists(String query, Tuple parameters, DBConn conn) {
     logger.debug("isRolloverExists:: Is rollover exists by query {}", query);
-    Promise<Boolean> promise = Promise.promise();
 
-    client.getPgClient()
-      .execute(query, parameters, reply -> {
-        if (reply.failed()) {
-          logger.debug("isRolloverExists:: Getting rollover by query {}", query, reply.cause());
-          handleFailure(promise, reply);
-        } else {
-          reply.result()
-            .spliterator()
-            .forEachRemaining(row -> promise.complete(row.get(Boolean.class, 0)));
-        }
-      });
-    return promise.future();
+    return conn.execute(query, parameters)
+      .map(rowSet -> rowSet.iterator().next().get(Boolean.class, 0))
```

Comment:
This logic is slightly rewritten with calling next(), is NPE or NoSuchElementException possible here now?

Reply:
I added a comment to clarify.